pages/dashboard/workspaces/[workspaceid]/collections/[collectionid]/settings/creators.vue

```diff
@@ -66,65 +76,131 @@ if (error.value) {
 }
 
 if (data.value) {
-  moduleData.creators = normalizeCreators(data.value);
+  creators.value = normalizeCreators(data.value);
 }
 
-const addCreator = () => {
-  moduleData.creators.push({
+const openAddCreatorDrawer = () => {
+  selectedCreator.value = {
     affiliation: "",
-    creatorIndex: moduleData.creators.length + 1,
+    creatorIndex: creators.value.length + 1,
     creatorName: "",
     familyName: "",
     givenName: "",
     identifier: "",
     identifierType: null,
     nameType: "Personal",
+  };
+
+  drawerAction.value = "Add";
+  showCreatorDrawer.value = true;
+};
+
+const openEditCreatorDrawer = (index: number) => {
+  const creator = creators.value[index];
+
+  if (creator) {
+    selectedCreator.value = {
+      affiliation: creator.affiliation,
+      creatorIndex: creator.creatorIndex,
+      creatorName: creator.creatorName,
+      familyName: creator.familyName,
+      givenName: creator.givenName,
+      identifier: creator.identifier,
+      identifierType: creator.identifierType,
+      nameType: creator.nameType,
+    };
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **issue (bug_risk):** Potential issue with creatorIndex assignment

Reassigning creatorIndex on save might lead to inconsistencies, especially if the creators array is manipulated between reads and writes. Ensure that creatorIndex remains stable and unique across operations.

```suggestion
+};
+
+const saveCreators = async () => {
+  const data = creators.value.map((creator, index) => {
+    return {
+      ...creator,
+      creatorIndex: index, // Ensure each creator has a stable, unique index based on their current position
```